app/scripts/modules/core/src/pipeline/config/stages/stage.html

```diff
@@ -100,7 +100,7 @@ <h4 ng-bind="stage.name || '[new stage]'"></h4>
     </page-section>
     <render-if-feature feature="artifacts">
       <page-section key="producesArtifacts" label="Produces Artifacts" visible="stageProducesArtifacts()">
-        <produces-artifacts stage="stage" pipeline="pipeline"></produces-artifacts>
+        <produces-artifacts-react stage="stage" pipeline="pipeline" on-produces-changed="producesArtifactsChanged" />
```

Comment:
Suggest wrapping this in a feature flag so that unflagged users get the existing experience.

Reply:
Done in 4dff56206.